### DISCUSSION TITLE
How to use ProLIF to obtain a ligand interaction network without mda result ?

### DISCUSSION THREAD

**Discussion by pkrezel** [chemosim-lab/ProLIF#130] [General]
HI, 

I would to create a ligand interaction network without results from molecular dynamic, just from pdb from PDB database, how can I do ?

**Comment by cbouy** [comment]
Hi @pkrezel,

If you're using a PDB file directly from the database, you'll need to add hydrogens so that they're all explicit before passing it to ProLIF.

First you have to split the protein and ligand components into separate PDB files.

Then you can start by adding explicit Hs to the protein. You can use webservers like [PypKa](https://pypka.org/run-pypka) or PropKa (also available as command line tools) for this matter.

You can then do the same for the ligand. If you have a SMILES string for your ligand, you could do something like this:
```python
from rdkit import Chem
from rdkit.Chem import AllChem

# load ligand pdb
lig_pdb = Chem.MolFromPDBFile("lig.pdb")
lig_smiles = Chem.MolFromSmiles(smiles_string)

# assign bond orders from the SMILES string
mol = AllChem.AssignBondOrdersFromTemplate(lig_smiles, lig_pdb)
# add hydrogens
mol = Chem.AddHs(mol)

# write SDFile
with Chem.SDWriter("lig.sdf") as w:
    w.write(mol)
```

Once that's done, you can do something like this:
```python
import prolif as plf
from prolif.plotting.network import LigNetwork

# load ligand
lig_suppl = plf.sdf_supplier("lig.sdf")

# load protein
prot_pdb = Chem.MolFromPDBFile("protonated_protein.pdb", removeHs=False)
prot = plf.Molecule(prot_pdb)

# run prolif
fp = plf.Fingerprint()
fp.run_from_iterable(lig_suppl, prot)
df = fp.to_dataframe()
print(df)

# visualize
df_atoms = fp.to_dataframe(return_atoms=True)
net = LigNetwork.from_ifp(
    df_atoms, lig_suppl[0],
    kind="frame", frame=0
)
net.show("pdb_lig_network.html")
```

**Comment by pkrezel** [comment]
Merci Cédric, je vais essayer.

Cordialement.

Pascal KREZEL

Le ven. 26 mai 2023 à 23:06, Cédric Bouysset ***@***.***> a
écrit :

> Hi @pkrezel <https://github.com/pkrezel>,
>
> If you're using a PDB file directly from the database
> Does your protein PDB file have explicit hydrogens on all atoms? It's a
> requirement for ProLIF to work correctly. If not, you can use webservers
> like PypKa <https://pypka.org/run-pypka> or PropKa (also available as
> command line tools) for this matter.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/chemosim-lab/ProLIF/discussions/130#discussioncomment-6015707>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFMDKG5W5OHCR5BB5MXB2ZLXIELMPANCNFSM6AAAAAAYL4TLYU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Pascal KREZEL
SCARTEK
Tél :06 86 77 49 27
Adresse: 2 route de Meung, Bizy, 41240 Ouzouer-le-Marché
